src/vm/docs/mod.rs

```diff
@@ -538,6 +540,96 @@ The `header-hash`, `burnchain-header-hash`, and `vrf-seed` properties return a 3
+functions are callable from other smart contracts, and may be invoked directly by users by submitting a transaction
+to the Stacks blockchain.
+
+Like other kinds of definition statements, `define-public` may only be used at the top level of a smart contract
+definition (i.e., you cannot put a define statement in the middle of a function body).
+
+Public functions _must_ return a ResponseType (using either `ok` or `err`). Any datamap modifications performed by
+a public function will be aborted if the function returns an `err` type. Public functions may be invoked by other
+contracts via `contract-call!`.",
+    example: "
+(define-public (hello-world (input int))
+  (begin (print (+ 2 input))
+         (ok input)))
+"
+};
+
+const DEFINE_PRIVATE_API: SpecialAPI = SpecialAPI {
+    name: "define",
+    input_type: "MethodSignature, MethodBody",
+    output_type: "",
+    signature: "(define (function-name (arg-name-0 arg-type-0) (arg-name-1 arg-type-1) ...) function-body)",
+    description: "`define` is used to define _private_ functions for a smart contract. Private
+functions may not be called from other smart contracts, nor may they be invoked directly by users.
+Instead, these functions may only be invoked by other functions defined in the same smart contract.
+
+Like other kinds of definition statements, `define` may only be used at the top level of a smart contract
+definition (i.e., you cannot put a define statement in the middle of a function body).
+
+Private functions may return any type.",
+    example: "
+(define (max-of (i1 int) (i2 int))
+  (if (> i1 i2)
+      i1
+      i2))
+(max-of 4 6) ;; returns 6
+"
+};
+
+const DEFINE_READ_ONLY_API: SpecialAPI = SpecialAPI {
+    name: "define-read-only",
+    input_type: "MethodSignature, MethodBody",
+    output_type: "",
+    signature: "(define-read-only (function-name (arg-name-0 arg-type-0) (arg-name-1 arg-type-1) ...) function-body)",
+    description: "`define-read-only` is used to define a _public read-only_ function for a smart contract. Such
+functions are callable from other smart contracts.
+
+Like other kinds of definition statements, `define-read-only` may only be used at the top level of a smart contract
+definition (i.e., you cannot put a define statement in the middle of a function body).
+
+Read-only functions may return any type. However, read-only functions
+may not perform any datamap modifications, or call any functions which
+perform such modifications. This is enforced both during type checks and during
+the execution of the function. Public read-only functions may
+be invoked by other contracts via `contract-call!`.",
+    example: "
+(define-read-only (just-return-one-hundred) 
+  (* 10 10))"
+};
+
+const DEFINE_MAP_API: SpecialAPI = SpecialAPI {
+    name: "define-map",
+    input_type: "MapName, KeyTupleDefinition, MapTupleDefinition",
+    output_type: "",
```

Comment:
For `output_type: does `user defined function` or "Not applicable" make sense?

Reply:
yeah, that would work

Reply:
switched to "Not Applicable"